### PR TITLE
Wait on sockets instead of sleep

### PR DESCRIPTION
**Pull request by beckje01**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -5,6 +5,12 @@ machine:
   environment:
     KAFKA_SERVER: 'localhost:9092'
 
+checkout:
+  post:
+    - mkdir -p ~/build-common
+    - curl -L https://smartthings-provisioning.s3.amazonaws.com/build-common/build-common.tar.gz | tar xvz -C ~/build-common
```
Comment (charliek) [inline circle.yml:11]: Did you make a OSS build of this?

Comment (beckje01) [inline circle.yml:11]: Nope it was already public

Comment (charliek) [inline circle.yml:11]: I thought we decided to not use this file for OSS stuff?

Comment (beckje01) [inline circle.yml:11]: I'll make an OSS version

**Pull request closed without merging**